packages/react-instantsearch/src/components/Highlighter.js

```diff
@@ -1,35 +1,79 @@
 import PropTypes from 'prop-types';
 import React from 'react';
+import classNames from './classNames';
+
+const cx = classNames('Highlight');
+
+function generateKey(i, value) {
+  return `split-${i}-${value}`;
+}
+
+export const Highlight = ({
```

Comment:
Should we export this as a component and then test the rendering out of it? So we freeze the implementation. Right now we are only testing that Highlight is called with some props, not the actual rendering of it (because it's shallow rendered in Highlighter.test.js)

Reply:
What do you mean by freezing the implementation? `Highlight` is tested separately so we don't need to do a full DOM rendering in `Highlighter`. We can just test that the the correct props are applied on the component. I think it's easier to reason about unit, rather than the all implementation.

Reply:
What I meant is that right no there's no test ensuring that the output of this single Highlight component is some specific DOM.

Which means that if we change the output without even touching the props being passed, nothing will break, while the public API contract will break (DOM will change).

If that's not clear still let me know

Reply:
Nope perfectly clear! But we already test the output of the `Highlight` component (see [#L11-39](https://github.com/algolia/react-instantsearch/blob/f3522ade97b94a2ee0c82590c02fc6c648be267d/packages/react-instantsearch/src/components/Highlighter.test.js#L11-L39))

Reply:
👍